### ISSUE TITLE
Request for additional packages in pyodide

### ISSUE THREAD

**Issue by dvvilkins** [pyodide/pyodide#4188]
## 🚀 Feature

Please include the following python packages in the next version of pyodide:
aiohttp
multidict
numexpr

### Motivation

These packages, which lack a pure python wheel due to C-extensions, are dependencies for me to preview my python script containing streamlit in VS Code. It would make my life so much easier if I could use stlite

### Pitch

Streamlit Preview VSCode Extension, powered by stlite, is a great way for me to develop streamlit webapps inside of VSCode. stlite depends on Pyodide. However, the python code I am using depends on packages that are not included in Pyodide, so i can't use the extension. Instead I have to use an external browser, which makes code development tedious.

### Alternatives

These package has C-extensions, and thus do not have a pure Python wheel on PyPI. I do not have the skills to package them myself.

### Additional context
Streamlit Preview VSCode Extension, powered by stlite
https://marketplace.visualstudio.com/items?itemName=whitphx.vscode-stlite

**Comment by agriyakhetarpal** [comment]
Both `aiohttp` and `multidict` are now available as a part of Pyodide, though `numexpr` isn't – a PR to add it would be welcome.

**Comment by hoodmane** [comment]
Also aiohttp doesn't really work.

**Comment by rajsite** [comment]
I'm in a similar-ish kind of boat. I'm trying to use stlite in a strict CORS environment (where access to pypi.org is blocked). I'm using the stlite docs that  describe how to point to a custom pyodide distribution: https://github.com/whitphx/stlite?tab=readme-ov-file#different-pyodide-distributions-pyodideurl-option

So I'm pulling the full pyodide release: https://github.com/pyodide/pyodide/releases/tag/0.27.6

But there are two packages not included in the full distribution that are pulling from pypi:

![Image](https://github.com/user-attachments/assets/0910f96b-c231-4a8f-a431-62f8761efbeb)

I'm struggling to find a way to include those packages in my self hosted copy of pyodide.

- Could be nice to include those packages in pyodide but that is a bit of a whack-a-mole as stlite updates and possibly adds more dependencies.
- Maybe I can create my own distribution. The closest docs I can find are more centered around [adding packages to the existing distribution](https://pyodide.org/en/stable/development/new-packages.html). Seems a bit rough to effectively fork to add a few pypi packages.
- Maybe there is a workflow for saving the wheels locally and mutating the pyodide-lock.json to reference them? I'm finding docs / examples of that challenging.
- Maybe do it in JS and run micropip against a local saved copy of the wheels manually? https://pyodide.org/en/stable/usage/loading-packages.html#installing-wheels-from-arbitrary-urls

**Comment by agriyakhetarpal** [comment]
Yes, a custom distribution usually works but you don't necessarily have to fork Pyodide to do that, as you've mentioned, @rajsite. You may use https://github.com/pyodide/pyodide-lock to manipulate the Pyodide lockfile, or add your own packages manually into it if you don't have too many of them to add. The tool doesn't receive a lot of attention, however, and there might be bugs. I wonder if we aren't documenting it in the appropriate locations.

**Comment by ryanking13** [comment]
> But there are two packages not included in the full distribution that are pulling from pypi:

I think you can host those packages somewhere else than PyPI, and install it through `pyodide.loadPackage("https://...")` or `micropip.install("...")`

**Comment by rajsite** [comment]
> I think you can host those packages somewhere else than PyPI, and install it through `pyodide.loadPackage("https://...")` or `micropip.install("...")`

That ended up working well! After playing with it I found stlite has a built-in mechanism to provide a set of packages to automatically `micropip` install as the `mount({requirements: string[]})` option. Can't find published docs but found the [type docs for requirements](https://github.com/whitphx/stlite/blob/3782652def614fd7cb50a65a9e6f6a7dc71502e9/packages/kernel/src/kernel.ts#L58). 

> You may use https://github.com/pyodide/pyodide-lock to manipulate the Pyodide lockfile, or add your own packages manually into it if you don't have too many of them to add.

Ahh, had not discovered that utility. Would be interesting to try. Would be handy if there was a node cli / library equivalent. I'm doing most of my configuring / setup of pyodide in a JS / node context. But could try it out!
